### ISSUE TITLE
Fail build ruby on last release (14332) zlib error

### ISSUE THREAD

**Issue by ddfznt** [Microsoft/BashOnWindows#317]
```
ddfznt@DED-WORKSPACE:~$ rbenv install 2.3.1
Downloading ruby-2.3.1.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.3/ruby-2.3.1.tar.bz2
Installing ruby-2.3.1...

BUILD FAILED (Ubuntu 14.04 using ruby-build 20160426-2-g82e275d)

Inspect or clean up the working tree at /tmp/ruby-build.20160502235511.12656
Results logged to /tmp/ruby-build.20160502235511.12656.log

Last 10 log lines:
Failed to configure tk/tkutil. It will not be installed.
configuring zlib
make[1]: Entering directory `/tmp/ruby-build.20160502235511.12656/ruby-2.3.1'
make -C ext/digest/sha2 -w --jobserver-fds=6,7 -j V= realclean
make[2]: Entering directory `/tmp/ruby-build.20160502235511.12656/ruby-2.3.1/ext/digest/sha2'
Makefile:39: *** missing separator.  Stop.
make[2]: Leaving directory `/tmp/ruby-build.20160502235511.12656/ruby-2.3.1/ext/digest/sha2'
make[1]: *** [ext/digest/sha2/realclean] Error 2
make[1]: Leaving directory `/tmp/ruby-build.20160502235511.12656/ruby-2.3.1'
make: *** [build-ext] Error 2

```

**Comment by benhillis** [comment]
Thanks for reporting.  I tried this locally and it looks like something is mangling the /tmp/ruby-build.20160502235511.12656/ruby-2.3.1/ext/digest/sha2/Makefile file.  Line 39 doesn't look well formed.  One of our developers actually just made a fix today that could potentially explain this.  I'll give this a try on tomorrow's build and let you know if that resolves this issue.

**Comment by luishmelo** [comment]
If you install [Brightbox's Ruby PPA](https://www.brightbox.com/docs/ruby/ubuntu/) package, and try to install some gems, you will get a bunch of other Makefile problems...

**Comment by simonbuchan** [comment]
@benhillis does that fix look like it could explain my bug #313 ?

**Comment by benhillis** [comment]
@simonbuchan Yes that's the bug I was referring to above.  The fix for that issue has been checked in so I'll give this a try today and see if it's unblocked.

**Comment by benhillis** [comment]
@ddfznt It looks like the bug fix I mentioned resolved the rbenv install issue:

```
root@BENHILL-VM7:~# rbenv install 2.3.1
Downloading ruby-2.3.1.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.3/ruby-2.3.1.tar.bz2
Installing ruby-2.3.1...
Installed ruby-2.3.1 to /root/.rbenv/versions/2.3.1
root@BENHILL-VM7:~/.rbenv/versions/2.3.1# ruby -v
rbenv: ruby: command not found

The `ruby' command exists in these Ruby versions:
  2.3.1

root@BENHILL-VM7:~/.rbenv/versions/2.3.1# cd bin
root@BENHILL-VM7:~/.rbenv/versions/2.3.1/bin# ./ruby -v
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]
```

**Comment by chathamws** [comment]
I have also come across this issue. Any idea when this fix will make it out into one of the preview builds? This is such exciting work. Looking forward to testing!

**Comment by ddfznt** [comment]
@chathamws "fast-ring" is not so fast. :/

**Comment by benhillis** [comment]
@chathamws Thank you for the kind words.  Unfortunately we don't have a firm timeline, keep an eye out for our release notes - we'll be sure to call out the "large files" fix in them.

**Comment by hut8** [comment]
I've been trying to install gems using the BrightBox Ruby PPA. A bunch of the Makefiles do, indeed, have random syntax errors in them. It wasn't resolved in the build that I installed today, but I look forward to the next one :)

**Comment by ddfznt** [comment]
@benhillis @hut8 Still failing in last release (14342).
https://gist.github.com/ddfznt/0089115b80acbd1acd04f0cb35ae4c2e

**Comment by ginolon** [comment]
@benhillis, how to fix it on my Windows Build Insider 14342? I need it to install rbenv. Plase.

**Comment by ginolon** [comment]
@ddfznt, any news?

**Comment by benhillis** [comment]
@ginolon - The fix for this issue did not make the 14342 build.  It will be in a future Insider build.

**Comment by ginolon** [comment]
Can you tell me what to ajdust by hand in my PC in the meantime?

**Comment by hut8** [comment]
@ginolon you can fix the Makefiles that are generated by hand, but then there are probably other errors caused by the same thing.

**Comment by ginolon** [comment]
@but why all this errors on this Ubuntu version for windows? Why? isn't it a real Ubuntu?

**Comment by aseering** [comment]
It's a real Ubuntu *except* that it's not running a real Linux kernel.  It's running on top of Windows directly, with some wrappers emulating the core low-level Linux system calls and other kernel functionality.  So you can install any Ubuntu package and run any Ubuntu program, but when the program tries to run, if that program tries to (broadly speaking) call some low-level function that hasn't been implemented correctly yet, you'll get an error.

If you want real Ubuntu on top of a real Linux kernel on top of Windows, that's called a Virtual Machine.  Lots of ways to do that today, and it works quite well.  But it doesn't integrate as nicely with your Windows environment.  That's the trade-off -- Windows and Linux represent a computer in slightly different ways from each other; so either you keep them a little bit separate (that's what a VM does), or you stitch them together extremely carefully (what this project aims to do).

**Comment by ginolon** [comment]
@aseering, I love you.

**Comment by ginolon** [comment]
Makefile:39 is creating problem also with "zlib".

**Comment by AbhimanyuAryan** [comment]
Is there a fix yet? I filled to install Rails with same problems :-1:

**Comment by ddfznt** [comment]
Fixed on 14352! <3

**Comment by AbhimanyuAryan** [comment]
Failed to install even in 14352

```
cooldudeabhi@ACERASPIRE:~$ rvm install 2.3.1
Searching for binary rubies, this might take some time.
Found remote file https://rubies.travis-ci.org/ubuntu/14.04/x86_64/ruby-2.3.1.tar.bz2
Checking requirements for ubuntu.
Requirements installation successful.
df: Warning: cannot read table of mounted file systems: No such file or directory
ruby-2.3.1 - #configure
ruby-2.3.1 - #download
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100 25.1M  100 25.1M    0     0  71285      0  0:06:10  0:06:10 --:--:-- 76063
No checksum for downloaded archive, recording checksum in user configuration.
ruby-2.3.1 - #validate archive
cat: /dev/fd/63: No such file or directory
cat: /dev/fd/63: No such file or directory
The downloaded package for https://rubies.travis-ci.org/ubuntu/14.04/x86_64/ruby-2.3.1.tar.bz2,
Does not contains single 'bin/ruby' or 'ruby-2.3.1',
Only '' were found instead.
Mounting remote ruby failed with status 4, trying to compile.
df: Warning: cannot read table of mounted file systems: No such file or directory
Checking requirements for ubuntu.
Requirements installation successful.
Installing Ruby from source to: /home/cooldudeabhi/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 93 13.7M   93 12.9M    0     0  86678      0  0:02:46  0:02:36  0:00:10 30547
```

**Comment by AbhimanyuAryan** [comment]
any idea @ddfznt?

**Comment by hut8** [comment]
It installed fine for me; it's just that the binary installation won't succeed because certain features that rvm requires for that part aren't implemented. Your paste looks like it was downloading from the source... did you let it finish?

**Comment by AbhimanyuAryan** [comment]
It worked finally I have received Windows Insider update today. On earlier built it didn't. Earlier I had some connectivity issue thanks for the support man :)

**Comment by AbhimanyuAryan** [comment]
Did you tried installing rails 5.0 @hut8?

**Comment by ddfznt** [comment]
@AbhimanyuAryan as mentioned here (https://github.com/rvm/rvm/commit/691ccb9fefd932d4af20b2e22e6501b87e8549a3), 
you need `sudo ln -sf /proc/self/fd /dev/fd`

**Comment by AbhimanyuAryan** [comment]
i  couldn't get to copy the error message but please have a lot at the errors in screenshot. `bundle install` doesn't succeed. Any Idea what went wrong? @ddfznt @hut8 

![1_error_bundle](https://cloud.githubusercontent.com/assets/8083613/15702951/72aa7044-2800-11e6-8bf5-217cf3c39533.PNG)
![2_error_bundle](https://cloud.githubusercontent.com/assets/8083613/15702952/744230fe-2800-11e6-809f-4e4bf44b8e44.PNG)
![3_error_bundle](https://cloud.githubusercontent.com/assets/8083613/15702954/75c056ae-2800-11e6-847d-d34080d56c01.PNG)

**Comment by ddfznt** [comment]
@AbhimanyuAryan This is due to the WSL permissions system.
Try running `sudo chmod 1777 ~/.bundle` and try again. (https://github.com/bundler/bundler/issues/4599#issuecomment-222202656)

**Comment by AbhimanyuAryan** [comment]
I still get those errors @ddfznt

**Comment by hut8** [comment]
@AbhimanyuAryan the fix is here https://github.com/Microsoft/BashOnWindows/issues/222#issuecomment-222317612 -- you've actually replied to that thread, so I'm not sure why there's confusion here.

This issue is about the Ruby build failing, and it's not failing for you anymore, and also the issue is closed, so for other bugs you encounter that actually have to do with WSL/Bash on Windows, you may want to actually file another issue.
